### PR TITLE
plotting ratio residuals

### PR DESCRIPTION
**Pull request by volodymyrss**
Additionally to sigma residuals it can be useful sometimes to see ratio of the residual to the model.

Sometimes because of complex dependency of detector efficiency on energy, the uncertainties change strongly and non-monotonically with energy. 
If there is a strong monotonic residual, it will appear to have features. The features correspond to sensitivity to the model, but they may be misinterpreted as line features.
Residual ratio does not have this problem (it has others)

### CODE REVIEW

**Comment by giacomov** [comment]
Thanks for this addition!

Can you post here an example on how to use it?

**Comment by volodymyrss** [comment]
Sure, here it is:

```python
threeML.display_spectrum_model_counts(jl,ratio_residuals=True)
```

File: threeML/io/plotting/data_residual_plot.py [hunk 1]
```diff
@@ -13,6 +13,9 @@ def __init__(self,**kwargs):
         :param kwargs:
         """
 
+        self.ratio_residuals = False
+        if 'ratio_residuals' in kwargs:
+            self.ratio_residuals = bool(kwargs.pop('ratio_residuals'))
```
Comment (grburgess) [inline threeML/io/plotting/data_residual_plot.py]: Can you make ratio_residuals a private attribute?

File: threeML/io/plotting/post_process_data_plots.py [hunk 2]
```diff
@@ -319,39 +323,45 @@ def display_spectrum_model_counts(analysis, data=(), **kwargs):
...
-                raise NotImplementedError("Not yet implemented")
+                    raise NotImplementedError("Not yet implemented")
 
         residual_plot.add_data(mean_energy,
                                new_rate / new_chan_width,
                                residuals,
+                               residual_yerr=residual_errors,
                                yerr=new_err / new_chan_width,
                                xerr=delta_energy,
                                label=data._name,
```
Comment (grburgess) [inline threeML/io/plotting/post_process_data_plots.py]: I think we should have a warning that ratios are not statistical quantites when using this option.

Comment (volodymyrss) [inline threeML/io/plotting/post_process_data_plots.py]: good point! Warning added. Feel free to reword.

Ratios are better than sigmas to asses relative shape of the residual structure, but there is no need to explain it to the user.

**Comment by giacomov** [comment]
@volodymyrss @grburgess can I merge this? It seems we have converged, but I want to make sure before merging.

**Comment by grburgess** [comment]
@giacomov looks good to me. Even if I don't like ratios :)

**Comment by volodymyrss** [comment]
So what's up with this one? It's marked as change requested, and the change was made long ago.

**Comment by grburgess** [comment]
That is my fault. Sorry.

**Comment by volodymyrss** [comment]
Thanks!

**Pull request merged**